app/furniture/marketplace/marketplace_component.rb

```diff
@@ -14,7 +14,7 @@ def render?
 
     def cart
       delivery_area = marketplace.delivery_areas.first if marketplace.delivery_areas.size == 1
-      @cart ||= carts.create_with(contact_email: shopper.person&.email, delivery_area: delivery_area).find_or_create_by(shopper: shopper, status: :pre_checkout)
+      @cart ||= marketplace.find_or_create_cart(shopper: shopper, delivery_area: delivery_area)
```

Comment:
This kind of reads as if the cart is being created for the `delivery_area`; rather than the default delivery area being set in the event a cart is created.

Perhaps we put line 16 in the `marketplace.find_or_create_cart(shopper: shopper)` implementation? Otherwise I think this is a bit better?